### ISSUE TITLE
Runtime crash when mixing app.draw and vulkan draw to frame

### ISSUE THREAD

**Issue by JoshuaBatty** [nannou-org/nannou#251]
It would be nice if we could mix simple nannou draw commands alongside vulkan graphics drawing to the same frame. 

For example, if I wanted to draw an ellipse using nannou on any of the vk_examples, i can do the following.

```
let draw = app.draw();
draw.ellipse()
       .x_y(app.mouse.x, app.mouse.y)
       .radius(300.0)
       .color(RED);
draw.to_frame(app, &frame).unwrap();    
```
If I place this code at the end of the vk_triangle example's view function then I get the following stack trace. 

```
thread 'main' panicked at 'assertion failed: `(left != right)`
  left: `Undefined`,
 right: `Undefined`', /Users/joshuabatty/.cargo/git/checkouts/vulkano-4e5b6de27e8f07b0/a237692/vulkano/src/command_buffer/sys.rs:1880:9
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:80
   2: std::panicking::default_hook::{{closure}}
             at src/libstd/sys_common/backtrace.rs:68
             at src/libstd/panicking.rs:210
   3: std::panicking::default_hook
             at src/libstd/panicking.rs:225
   4: <std::panicking::begin_panic::PanicPayload<A> as core::panic::BoxMeUp>::get
             at src/libstd/panicking.rs:488
   5: std::panicking::continue_panic_fmt
             at src/libstd/panicking.rs:395
   6: std::panicking::try::do_call
             at src/libstd/panicking.rs:350
   7: vulkano::command_buffer::sys::UnsafeCommandBufferBuilderPipelineBarrier::add_execution_dependency
             at /Users/joshuabatty/.cargo/registry/src/github.com-1ecc6299db9ec823/conrod_vulkano-0.62.0/<::std::macros::panic macros>:8
   8: vulkano::command_buffer::synced::base::CbKey::conflicts_buffer
             at /Users/joshuabatty/.cargo/git/checkouts/vulkano-4e5b6de27e8f07b0/a237692/vulkano/src/command_buffer/synced/base.rs:585
   9: core::fmt::ArgumentV1::new
             at /Users/joshuabatty/.cargo/git/checkouts/vulkano-4e5b6de27e8f07b0/a237692/vulkano/src/command_buffer/synced/commands.rs:120
  10: <std::sys::unix::os_str::Buf as core::clone::Clone>::clone
             at /Users/joshuabatty/.cargo/git/checkouts/vulkano-4e5b6de27e8f07b0/a237692/vulkano/src/command_buffer/auto.rs:560
  11: nannou::frame::AddCommands::begin_render_pass::{{closure}}
             at src/frame.rs:287
  12: <nannou::draw::backend::vulkano::vs::Layout as core::clone::Clone>::clone
             at src/frame.rs:261
  13: <nannou::draw::backend::vulkano::vs::Layout as core::clone::Clone>::clone
             at src/frame.rs:287
  14: nannou::frame::AddCommands::draw_indexed::{{closure}}
             at src/draw/backend/vulkano.rs:370
  15: nannou::app::update_interval
             at src/app.rs:1084
  16: vk_triangle::view
             at examples/vulkan/vk_triangle.rs:195
  17: nannou::app::view_frame
             at ./src/app.rs:1999
  18: nannou::app::run_loop_mode_refresh_sync
             at ./src/app.rs:1474
  19: nannou::app::run_loop
             at ./src/app.rs:1162
  20: <nannou::app::Builder<M, E>>::run
             at ./src/app.rs:569
  21: vk_triangle::main
             at examples/vulkan/vk_triangle.rs:17
  22: std::rt::lang_start::{{closure}}
             at /rustc/14b96659e4e9b2f31431df48f47c219957e2666a/src/libstd/rt.rs:74
  23: std::panicking::try::do_call
             at src/libstd/rt.rs:59
             at src/libstd/panicking.rs:307
  24: panic_unwind::dwarf::eh::read_encoded_pointer
             at src/libpanic_unwind/lib.rs:102
  25: std::panicking::update_count_then_panic
             at src/libstd/panicking.rs:286
             at src/libstd/panic.rs:398
             at src/libstd/rt.rs:58
  26: std::rt::lang_start
             at /rustc/14b96659e4e9b2f31431df48f47c219957e2666a/src/libstd/rt.rs:74
  27: vk_triangle::model::scope::attachment_layouts
```

**Comment by mitchmindtree** [comment]
Possibly fixed by #242.

**Comment by mitchmindtree** [comment]
Just tested this on Joshy's lappy and it is indeed fixed!
